app/controllers/application_controller/ci_processing.rb

```diff
@@ -872,10 +872,13 @@ def discover
             end
             Host.discoverByIpRange(from_ip, to_ip, options)
           else
-            if params[:discover_type_selected] == ExtManagementSystem.ems_cloud_discovery_types['azure']
-              ManageIQ::Providers::Azure::CloudManager.discover_queue(@client_id, @client_key, @azure_tenant_id)
+            cloud_manager = ManageIQ::Providers::CloudManager.subclasses.detect do |ems|
+              ems.supports_discovery? && ems.ems_type == params[:discover_type_selected]
+            end
+            if cloud_manager.ems_type == 'azure'
```

Comment:
> why this check for Azure?
because azure and amazon handle authentication slightly different. To unify this @juliancheal is working on refactoring the authentication part of the cloud controllers